relay/kusama/src/lib.rs

```diff
@@ -874,9 +867,11 @@ pub struct TreasuryBurnHandler;
 
 impl OnUnbalanced<BalancesNegativeImbalance> for TreasuryBurnHandler {
 	fn on_nonzero_unbalanced(amount: BalancesNegativeImbalance) {
-		if let Some(TreasuryBurnParameters { account, .. }) =
-			dynamic_params::treasury::BurnParameters::get()
-		{
+		let portion = dynamic_params::treasury::BurnPortion::get();
+		let account = dynamic_params::treasury::BurnDestination::get();
+
+		if !portion.is_zero() && account.0.is_some() {
+			let account = account.0.expect("given `account.0.is_some`; qed");
 			// Must resolve into existing but better to be safe.
 			Balances::resolve_creating(&account, amount);
```

Comment:
you can refactor to avoid the `expect`:

```suggestion
			if let Some(account) = account.0.as_ref() {
				// Must resolve into existing but better to be safe.
				Balances::resolve_creating(account, amount);
			}
```

no need for else branch since you know it will always be `Some`